packages/react-native/src/storage/StreamWriter.ts

```diff
@@ -17,3 +19,56 @@ export interface StreamWriter {
      */
     close(key: string): boolean;
 }
+
+export class FileWritableStream extends WritableStream {
+    constructor(
+        public readonly path: string,
+        streamWriter: StreamWriter,
+    ) {
+        super(new NativeUnderlyingSink(path, streamWriter));
+    }
+}
+
+export class NativeUnderlyingSink implements UnderlyingSink<string> {
+    private _streamId?: string;
+
+    constructor(
+        public readonly path: string,
+        private readonly _streamWriter: StreamWriter,
+    ) {}
+
+    public async start() {
+        this._streamId = this._streamWriter.create(this.path);
+        if (!this._streamId) {
+            throw new Error(`Failed to open file ${this.path}.`);
+        }
+    }
+
+    public close() {
+        if (!this._streamId) {
+            return;
+        }
+
+        if (!this._streamWriter.close(this._streamId)) {
+            throw new Error(`Failed to close file ${this.path}.`);
+        }
+    }
+
+    public async write(chunk: string) {
+        if (!this._streamId) {
+            throw new Error('File is not open.');
+        }
+
+        if (!(await this._streamWriter.append(this._streamId, chunk))) {
+            throw new Error(`Failed to write data to file ${this.path}.`);
```

Comment:
what benefits bring throwing an exception here? Should we fail silently? In theory our library should never throw an error. Is there any consequence if we fail silently? If we fail loud, what the user can do to fix it?

Reply:
We shouldn't fail silently here. However, I added `catch` to `write` in `FileBreadcrumbStorage` and `close` in `FileChunkSink`.